### PR TITLE
8273346: Expand library mappings to IEEE 754 operations

### PR DESCRIPTION
**Pull request by jddarcy**
Generally add apiNote's to map from Java library methods to particular IEEE 754 operations. For now, I only added such notes to java.lang.Math and not java.lang.StrictMath.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.java.net/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8273346](https://bugs.openjdk.org/browse/JDK-8273346): Expand library mappings to IEEE 754 operations

### Reviewers
 * [Brian Burkhalter](https://openjdk.java.net/census#bpb) (@bplb - **Reviewer**) ⚠️ Review applies to [ba3e9a75](https://git.openjdk.java.net/jdk/pull/8876/files/ba3e9a75831f1c93f00c49bb83ee3973549f14b3)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/8876/head:pull/8876` \
`$ git checkout pull/8876`

Update a local copy of the PR: \
`$ git checkout pull/8876` \
`$ git pull https://git.openjdk.java.net/jdk pull/8876/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 8876`

View PR using the GUI difftool: \
`$ git pr show -t 8876`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/8876.diff">https://git.openjdk.java.net/jdk/pull/8876.diff</a>

</details>

### CODE REVIEW

File: src/java.base/share/classes/java/math/RoundingMode.java [hunk 1]
```diff
@@ -39,6 +39,21 @@
...
  * an absolute value greater than one.
  *
+ * <p>More generally, a rounding policy defines a mapping from the
+ * real numbers to a subset of representable values. In the case of
+ * {@link BigDecimal}, the representable values are a function of the
+ * {@linkplain MathContext#getPrecision() precision} being used in the
+ * computation. Assuming the mathematical result is within the
+ * exponent range of {@code BigDecimal}, the mathematical result will
+ * be exactly representable in the result precision or fall between
+ * two representable values. In the case of falling between two
```
Comment (rgiulietti) [inline src/java.base/share/classes/java/math/RoundingMode.java:49]: Perhaps better would be
`two adjacent representable values.`

File: src/java.base/share/classes/java/lang/Math.java [hunk 2]
```diff
@@ -750,6 +765,10 @@ public static double pow(double a, double b) {
      * equal to the value of {@code Integer.MAX_VALUE}, the result is
      * equal to the value of {@code Integer.MAX_VALUE}.</ul>
      *
+     * @apiNote
+     * This method corresponds to the convertToIntegerTiesToEven
+     * operation defined in IEEE 754.
+     *
```
[the remark this answers is not in the document]
Comment (jddarcy) [reply src/java.base/share/classes/java/lang/Math.java:771]: Good catch; found a different 754 operation that seems to fit.

File: src/java.base/share/classes/java/math/RoundingMode.java (hunk 1 above)
Comment (jddarcy) [reply src/java.base/share/classes/java/math/RoundingMode.java]: Agreed; adjusted in subsequent push.

**Comment by jddarcy** [comment]
I look another pass over the methods in Float, Double, and Math for opportunities to link to IEEE 754 operations. At least for now, I choose not to link to abs since IEEE 754 abs can have different behavior for NaN values that java.lang.Math.abs. If a link is made, this difference would need to be noted.

File: src/java.base/share/classes/java/lang/Math.java [hunk 3]
```diff
@@ -819,8 +819,8 @@ public static int round(float a) {
      * equal to the value of {@code Long.MAX_VALUE}.</ul>
      *
      * @apiNote
-     * This method corresponds to the convertToIntegerTiesToEven
-     * operation defined in IEEE 754.
+     * This method corresponds to the convertTowardPositive operation
+     * defined in IEEE 754.
```
Comment (rgiulietti) [inline src/java.base/share/classes/java/lang/Math.java:823]: Probably what is meant is IEEE `convertToIntegerTowardPositive` (not `convertTowardPositive`).
However, that's another kind of rounding, which always rounds any non-integral value toward positive infinity, not just ties. This is not what happens with this method.

Comment (rgiulietti) [reply src/java.base/share/classes/java/lang/Math.java:823]: > not just ties
Plz. disregard this 3 words

Comment (jddarcy) [reply src/java.base/share/classes/java/lang/Math.java:823]: Okay, checking the IEEE 754-2019 spec again, there are operations to round a floating-point value to an integral-valued floating-point value (roundToIntegral{$ROUNDING_DIRECTION_ATTRIBUTE} and separately operations to round from a floating-point value to a value in an integer format (convertToInteger{$ROUNDING_DIRECTION_ATTRIBUTE} ). However, as noted, there is no IEEE 754 operations for rounding up only for ties. I'll remove the API notes for the round methods; thanks for catching this.

File: src/java.base/share/classes/java/math/RoundingMode.java [hunk 4]
```diff
@@ -39,6 +39,21 @@
...
+ * {@link BigDecimal}, the representable values are a function of the
+ * {@linkplain MathContext#getPrecision() precision} being used in the
+ * computation. Assuming the mathematical result is within the
+ * exponent range of {@code BigDecimal}, the mathematical result will
+ * be exactly representable in the result precision or will fall between
+ * two adjacent representable values. In the case of falling between two
+ * representable values, the rounding policy determines which of those
+ * two bracketing values is the result. For in-range real numbers, for
+ * a given set of representable values, a rounding policy maps a
+ * continuous segment of real number line to a single representable
```
Comment (bplb) [inline src/java.base/share/classes/java/math/RoundingMode.java:53]: "the" real number line

**Review by bplb — APPROVED** [review]
Approved assuming comments will be addressed prior to integration.

**Comment by rgiulietti** [comment]
LGTM

**Pull request closed without merging**